lib/Conversion/HandshakeToDC/HandshakeToDC.cpp

```diff
@@ -308,16 +308,28 @@ struct UnitRateConversionPattern : public ConversionPattern {
   matchAndRewrite(Operation *op, ArrayRef<Value> operands,
                   ConversionPatternRewriter &rewriter) const override {
     llvm::SmallVector<Value, 4> inputData;
-    llvm::SmallVector<Value, 4> inputTokens;
-    for (auto input : operands) {
-      auto dct = unpack(rewriter, input);
-      inputData.push_back(dct.data);
-      inputTokens.push_back(dct.token);
+    Value outToken;
+    if (operands.empty()) {
+      if (!op->hasTrait<OpTrait::ConstantLike>())
+        return op->emitOpError(
+            "no-operand operation which isn't constant-like. Too dangerous "
+            "to assume semantics - won't convert");
+
+      // Constant-like operation; assume the token can be represented as a
+      // constant `dc.source`.
+      outToken = rewriter.create<dc::SourceOp>(op->getLoc());
+    } else {
+      llvm::SmallVector<Value, 4> inputTokens;
```

Comment:
```suggestion
     SmallVector<Value> inputTokens;
```
Nit: I would not specify a size if there is no particular reason. I rather suggest to call reserve for such cases, as this ensures that there will be at most one heap allocation.